### PR TITLE
embedding server templates

### PR DESCRIPTION
**Pull request by colinhoglund**
This copies the html templates from the existing auth portal and sets up the basic functionality to embed them into the server package using https://pkg.go.dev/embed.

Use `make docker-run` to test this locally on `localhost:8080`.
or
Run from the public registry: `docker run -p 8080:8080 public.ecr.aws/kanopy-sandbox/k8s-auth-portal:latest`

This is publishing to the sandbox public ECR registry for now. Also, this is using a fork of the `kaniko-ecr` plugin with support for repo creation. There is an upstream PR open for this: https://github.com/drone/drone-kaniko/pull/24

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 import (
-	"fmt"
+	"embed"
+	"html/template"
 	"net/http"
+
+	log "github.com/sirupsen/logrus"
 )
 
+//go:embed templates
```
Comment (dskatz) [inline internal/server/server.go:11]: Very nifty!

File: internal/server/server_test.go [hunk 2]
```diff
@@ -1,28 +1,23 @@
...
+}
 
-		if w.Code != 200 {
-			t.Error("test failed")
-		}
-	}
+func TestHandleRoot(t *testing.T) {
+	w := httptest.NewRecorder()
+	s.ServeHTTP(w, httptest.NewRequest("GET", "/", nil))
+	assert.Equal(t, http.StatusOK, w.Code)
```
Comment (dskatz) [inline internal/server/server_test.go:22]: Is the plan to expand these tests as new features are ported over from the original auth portal?

**Review by dskatz — DISMISSED** [review]
LGTM, overall. One minor question that doesn't block this PR.

File: internal/server/server_test.go (hunk 2 above)
Comment (colinhoglund) [reply internal/server/server_test.go:22]: Yes, new tests can be added as the routes are updated.

**Review by dskatz — APPROVED** [review]
LGTM

**Pull request merged**